packages/odyssey-react-mui/src/labs/SideNav/SideNav.tsx

```diff
@@ -121,16 +132,103 @@ const SideNavFooterContainer = styled("div", {
   },
 }));
 
+const getHasScrollableContent = (scrollableContainer: HTMLElement) =>
+  scrollableContainer.scrollHeight > scrollableContainer.clientHeight;
+
 const SideNav = ({
   navHeaderText,
   isCollapsible,
   onCollapse,
   onExpand,
   sideNavItems,
+  expandedWidth = "300px",
   footerItems,
+  footerComponent,
+  logo,
 }: SideNavProps) => {
   const [isSideNavCollapsed, setSideNavCollapsed] = useState(false);
   const odysseyDesignTokens = useOdysseyDesignTokens();
+  const [isContentScrollable, setIsContentScrollable] = useState(false);
+  const [hasContentScrolled, setHasContentScrolled] = useState(false);
+  const scrollableContentRef = useRef<HTMLUListElement>(null);
+  const resizeObserverRef = useRef<ResizeObserver | null>(null);
+  const intersectionObserverRef = useRef<IntersectionObserver | null>(null);
+
+  useEffect(() => {
+    const setWhetherContentIsScrollable = () => {
+      if (
+        scrollableContentRef.current &&
+        scrollableContentRef.current.parentElement
+      ) {
+        setIsContentScrollable(
+          getHasScrollableContent(scrollableContentRef.current.parentElement),
+        );
+      }
+    };
+
+    // If the window is resized, we may need to re-determine if the scrollable container has overflow
+    // Setup a ResizeObserver to know if the size of the scrollableContent changes
+    let resizeObserverDebounceTimer: ReturnType<typeof requestAnimationFrame>;
+    if (!resizeObserverRef.current) {
+      resizeObserverRef.current = new ResizeObserver(() => {
+        cancelAnimationFrame(resizeObserverDebounceTimer);
+        resizeObserverDebounceTimer = requestAnimationFrame(
+          setWhetherContentIsScrollable,
+        );
+      });
```

Comment:
I think we want to create this function outside of `new ResizeObserver()` because I'm sure you _also_ want to run it at mount rather than just when a resize occurs. If someone starts with their browser too short, then you'd want this to execute immediately.

Reply:
Oh wait, we're calling it already lower down. I didn't notice with all the PR comments.

Gonna leave my comment here in case I missed something, but we should be fine!